### PR TITLE
Rails - Forms

### PR DESCRIPTION
**Pull request by ybadmus**
This project illustrates multiple ways to build a form in rails that can send parameters to a controller so that it can build an instance of a model with those parameters.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,24 +1,86 @@
...
+- Press the "Code" button and get the repo link.
+- Clone it using git.
+- Open terminal on that folder.
+- run `yarn install` 
+- run `npm install`
+- run `rails webpacker:install`
+- run `bundle install`
+- run `rails db: migrate` to migrate the database 
+- run `rails server`
+- Go to `http://localhost:3000` on your browser to view the homepage of the application
```
Comment (whiz25) [inline README.md:28]: - Add instruction for accessing the sign up page. `localhost:3000/users/new`
- Add the path for editing user. `localhost:3000/users/{id}/edit`

File: app/views/users/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,15 @@
+<h1>Create a new user</h1>
```
Comment (whiz25) [inline app/views/users/new.html.erb:1]: - There should be two commented out forms before this one. Please add the two commented out forms below.
#### `form`
```ruby
<form action="users_path" method="post" accept-charset="UTF-8">
  <label for="user[username]">Username</label>
  <input name="user[username]" type="text" />
  <input
    type="hidden"
    name="authenticity_token"
    value="<%= form_authenticity_token %>"
  />
  <input type="submit" value="Submit" />
</form>
```

#### `form_tag`
```ruby
<%= form_tag(users_path) do %> <%= label_tag 'username' %> <%= text_field_tag
'username' %> <%= label_tag 'email' %> <%= email_field_tag 'email' %> <%=
label_tag 'password' %> <%= password_field_tag 'password' %> <%= submit_tag %>
<% end %>
```
### Note
- The aim of this PR is to make sure that you know who to create a form using one of the three ways

**Review by whiz25 — CHANGES_REQUESTED** [review]
## Changes Required :pray:
Nice work on this PR. You are on the right path to success: no linter errors, readme file looks professional, a user can signup. Well done!
Let's however make some changes so we can merge this PR.
- [ ] Unable to edit an existing user. I visited `localhost:3000/users/1/edit` after signing up to edit my info but got this error `No route matches [POST] "/users/1/edit"` after hitting the `Submit` button. Please fix that error so a user can edit their details. Also, it would be a good idea to prefill the forms with the existing user's data like email.
- [ ] Check my comments for suggestions on how to make this PR better.


### Martin Afani

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -0,0 +1,35 @@
+class UsersController < ApplicationController
+  def new
+    @user = User.new
+  end
+
+  def create
+    # @user = User.new(username: params[:username], email: params[:email], password: params[:password])
```
Comment (enaburekhan) [inline app/controllers/users_controller.rb]: Remove commented-out code. It makes it difficult to understand the codebase and also looks untidy.

File: app/views/users/new.html.erb [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+</form>-->
+
+<%= #form_tag(users_path) do %> <%= label_tag 'username' %>
+  <%= #text_field_tag 'username' %>
+  <%= #label_tag 'email' %>
+  <%= #email_field_tag 'email' %>
+  <%= #label_tag 'password' %>
+  <%= #password_field_tag 'password' %>
+  <%= #submit_tag %>
+<% #end %>
```
Comment (enaburekhan) [inline app/views/users/new.html.erb]: This method of commenting out is generating an error. Instead, do it this way:
<%#= form_tag(users_path) do %> <%#= label_tag 'username' %>
  <%#= text_field_tag 'username' %>
  <%#= label_tag 'email' %>
  <%#= email_field_tag 'email' %>
  <%#= label_tag 'password' %>
  <%#= password_field_tag 'password' %>
  <%#= submit_tag %>
<%# end %>

**Review by enaburekhan — CHANGES_REQUESTED** [review]
# Status: Changes requested:full_moon:
**Hi team!
Thank you for implementing all the changes requested by the last reviewer:clap::clap:**.

**Major highlights**

- It's possible to create a new user in app.
- It's also possible to edit the existing user
- user_params method requires the correct fields.

However, we still need to make a few changes to get the project approved:muscle:.

**To improve**

- As already highlighted in the PR below. Remove commented-out code from the user controller.
- Check my comment on new.html.erb in view and fix it.

[ Optional]

- form view to displays a list of the error messages (with #errors and full_messages usage) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_extra_shared_error.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/forms))  
- validate the email format by using either a regular expression or a built in URI method.

Make the required changes and when you are ready, request another review.
Kindly drop your questions :point_down: and ping me on slack @ Eric Enaburekhan to respond.
Happy coding:blush:
Eric Enaburekhan

**Review by ioilmio — APPROVED** [review]
# Helllo @JuanDavid31 @ybadmus  👋🏽 

# Your project satisfy all the minimum requirements so it is approved 🎉  Congratulations

### Change the title in the readme, is saying Enumerable methods right now
### For your next project be sure to have your main and development branch totally empty at the beginning, there's already rails code in them.

*For any doubts or info you can contact me on slack @illuminatosalvatore*

**Pull request merged**